### DISCUSSION TITLE
Política e-greedy

### DISCUSSION THREAD

**Discussion by fernandodediegoavila** [JavierBZ/Tarea3-IIC3697#2] [Q&A] [answered]
Hola!

En el código de la ayudantía, se incluye un llamado a la función exploration (que tenemos que definir), la cual es la encargada de llevar a cabo la política e-greedy, la cual alcanza valor 0.1 después de 1 millón de iteraciones. Tengo una duda con respecto a la implementación entregada, ya que exploration está en función de t, con t el número de iteraciones de una época, pero t solo alcanzará el valor 250 mil, que es el largo de una época, por lo que la política e-greedy nunca alcanzará a llegar a 0.1.
No deberíamos dejar exploration en función de otr contador, que indique el número total de iteraciones, a través de todas las épocas?

Muchos saludos!!

**Comment by JavierBZ** [comment, marked as answer]
Hola, tienes razón, el t de la función debe ser un contador fuera de las épocas para hacer la política e-greedy correctamente.

Recuerden que el código es más una guía y pueden haber más cosas aparte de las comentadas que deban modificar un poco.

Saludos!

**Reply by fernandodediegoavila** [reply]
Buenísimo, muchas gracias Javier.

**Reply by danustilovsky** [reply]
Hola @JavierBZ!

Me surgio una duda con respecto a el t que se debe tomar en cuenta para esto. Debe ser la cantidad de steps totales, o solo la cantidad de steps que han pasado mientras el modelo se está entrenando? 

En otras palabras debería ser  epoch * STEPS_PER_EPOCH + t  o epoch * STEPS_PER_EPOCH+ t  - learning_starts

Gracias!

**Reply by JavierBZ** [reply]
Hola!

Es desde el inicio de todo el entrenamiento, considerando la exploración inicial de 50.000

Saludos
